### PR TITLE
Attributes as functions

### PR DESCRIPTION
**Pull request by stephencelis**
Phantom types were a nice, lightweight way of adding type-safety, but had a couple disadvantages:

  - Variables can't be overloaded in Swift (so we can't support `style` taking both `String` and `CSS` types.

  - They added an extra operator, `=>`.

### CODE REVIEW

File: Sources/HTML/Attributes.swift [hunk 1]
```diff
@@ -4,39 +4,130 @@ public enum Method: Value {
...
-public let readonly = AttributeKey<Bool>(key: "readonly")
-public let required = AttributeKey<Bool>(key: "required")
-public let rows = AttributeKey<Int>(key: "rows")
-public let selected = AttributeKey<Bool>(key: "selected")
-public let src = AttributeKey<String>(key: "src")
-public let step = AttributeKey<Int>(key: "step")
-public let type = AttributeKey<String>(key: "type")
-public let value = AttributeKey<String>(key: "value")
-public let width = AttributeKey<Int>(key: "width")
+public func action(_ action: String) -> Attribute {
```
Comment (stephencelis) [inline Sources/HTML/Attributes.swift]: Really wish there were a lighter-weight syntax for creating real functions in Swift, but this isn't too bad.

File: Tests/HTMLTests/FullDocumentTests.swift [hunk 2]
```diff
@@ -20,21 +20,21 @@ class FullDocumentTests: XCTestCase {
           ]
         ),
         body(
-          [ HTML.class => "home",
-            id => "home"
+          [ HTML.class <| "home",
+            id <| "home"
```
Comment (stephencelis) [inline Tests/HTMLTests/FullDocumentTests.swift]: Now that these are functions, we can reuse our function application operator instead! Could even call functions directly if we prefer:

``` swift
[ HTML.class("home"),
  id("home")
],
```

File: Sources/HTML/Attributes.swift [hunk 3]
```diff
@@ -4,39 +4,130 @@ public enum Method: Value {
...
+
+public func src(_ src: String) -> Attribute {
+  return Attribute("src", src)
+}
+
+public func step(_ step: Int) -> Attribute {
+  return Attribute("step", step)
+}
+
+public func style(_ style: String) -> Attribute {
```
Comment (stephencelis) [inline Sources/HTML/Attributes.swift]: Now we can support string content for style tags!

Comment (mbrandonw) [inline Sources/HTML/Attributes.swift]: and third party CSS libs!

**Pull request merged**